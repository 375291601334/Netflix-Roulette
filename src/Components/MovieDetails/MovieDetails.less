@import '../../Styles/variables.less';

.wrapper {
  display: flex;
  flex-direction: column;
  padding: 20px 100px;
  background: linear-gradient(rgba(0, 0, 0, 0.9), rgba(0, 0, 0, 0.9)), url(../../../assets/img/films.jpeg);
  color: @white;

  .tablet({
    padding: 20px 50px 50px;
  });

  .mobile({
    padding: 20px 30px 30px;
  });

  h2 {
    display: inline-block;
    font-size: 40px;
    margin-right: 40px;
  }
}

.searchIcon {
  display: inline-block;
  align-self: flex-end;
  mask-image: url(../../../assets/icons/loupe.svg);
  -webkit-mask-image: url(../../../assets/icons/loupe.svg);
  mask-size: 15px 20px;
  -webkit-mask-size: 15px 20px;
  width: 15px;
  height: 20px;
  background: @red;
  cursor: pointer;

  &:hover {
    filter: contrast(190%);
    mask-size: 35px 40px;
    -webkit-mask-size: 35px 40px;
    width: 35px;
    height: 40px;
    margin-bottom: -20px;
  }
}

.movieInfo {
  display: flex;
  flex-wrap: nowrap;
  margin: 30px 0;

  > *:not(:last-child) {
    margin-right: 40px;
  }

  .mobile({
    flex-direction: column;

    > *:not(:last-child) {
      margin-bottom: 40px;
    }
  });
}

.poster {
  width: 40%;
  max-height: 400px;
  background-color: @darkGreyTransparent;

  .mobile({
    width: 100%;
    min-height: 400px;
  });
}

.mainInfoContainer {
  width: 60%;

  > *:not(:last-child) {
    margin-bottom: 20px;
  }

  .mobile({
    width: 100%;
  });
}

.numberCharacteristics {
  color: @red;
  font-size: 20px;

  > *:not(:last-child) {
    margin-right: 40px;
  }
}

.votes {
  display: inline-block;
  width: 60px;
  height: 60px;
  padding: 15px 0 0 0;
  font-size: 30px;
  font-family: @MontserratBoldFont;
  border: 1px solid @white;
  box-sizing: border-box;
  border-radius: 50%;
  text-align: center;
  color: green;
}
