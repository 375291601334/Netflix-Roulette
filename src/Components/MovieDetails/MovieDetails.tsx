import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { useRouter } from 'next/router';
import Link from 'next/link';
import { Movie } from '../MovieCard';
import css from './MovieDetails.less';

export function MovieDetails({ getMovie }: { getMovie: (id: string) => Movie}) {
  const [movie, setMovie] = useState<Movie>((undefined as unknown) as Movie);
  const router = useRouter();
  const { id } = router.query;

  useEffect(
    () => {
      const selectedMovie = getMovie(id as string);
      setMovie(selectedMovie);
    },
    [id],
  );

  const posterStyle = {
    backgroundSize: 'contain',
    backgroundRepeat: 'no-repeat',
    backgroundPosition: 'center',
    backgroundImage: `url(${movie?.poster_path})`,
  };

  return (
    <div className={css.wrapper}>
      <Link href={'/films'}><span className={css.searchIcon} /></Link>
      <div className={css.movieInfo}>
        <div className={css.poster} style={posterStyle} />
        <div className={css.mainInfoContainer}>
          <h2>{movie?.title}</h2>
          <span className={css.votes}>{movie?.vote_average}</span>
          <p>{movie?.tagline}</p>
          <div className={css.numberCharacteristics}>
            { movie?.release_date && <span>{new Date(movie.release_date).getFullYear()}</span>}
            <span>{movie?.runtime} min</span>
          </div>
          <div>{movie?.overview}</div>
        </div>
      </div>
    </div>
  );
}

MovieDetails.propTypes = {
  getMovie: PropTypes.func.isRequired,
};
