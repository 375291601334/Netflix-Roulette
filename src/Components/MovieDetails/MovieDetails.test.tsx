import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { MovieDetails } from './MovieDetails';

const movie = {
  id: 123,
  title: 'Movie title',
  tagline: '',
  vote_average: 7.3,
  vote_count: 8.5,
  release_date: '2020-10-13',
  poster_path: 'https://someImage.com',
  overview: 'Movie overview',
  budget: 4365600,
  revenue: 5466453,
  genres: ['Comedy'],
  runtime: 123,
};

const getMovie = jest.fn().mockReturnValue(movie);

describe('MovieDetails', () => {
  test('render MovieDetails', () => {
    const { baseElement } = render(
      <BrowserRouter>
        <MovieDetails getMovie={getMovie} />
      </BrowserRouter>,
    );
    expect(baseElement).toMatchSnapshot();
  });
});
