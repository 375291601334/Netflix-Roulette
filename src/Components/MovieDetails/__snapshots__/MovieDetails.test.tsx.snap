// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails render MovieDetails 1`] = `
<body>
  <div>
    <div
      class="wrapper"
    >
      <span
        class="searchIcon"
      />
      <div
        class="movieInfo"
      >
        <div
          class="poster"
          style="background-size: contain; background-repeat: no-repeat; background-position: center; background-image: url(https://someImage.com);"
        />
        <div
          class="mainInfoContainer"
        >
          <h2>
            Movie title
          </h2>
          <span
            class="votes"
          >
            7.3
          </span>
          <p />
          <div
            class="numberCharacteristics"
          >
            <span>
              2020
            </span>
            <span>
              123
               min
            </span>
          </div>
          <div>
            Movie overview
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
